### ISSUE TITLE
Publish old jade versions to npmjs under pug

### ISSUE THREAD

**Issue by iknowcss** [pugjs/pug#2384]
Sorry if this has been raised already. I checked the issues but couldn't find something matching this request.

Some other projects (e.g. mocha) using old versions of jade will break once the rename is complete. Will there be a mass re-publish of old versions so that those projects can reference pug instead of jade in their package.jsons?

Here is a comment from a mocha.js issue about the warning in npm:
https://github.com/mochajs/mocha/issues/2200#issuecomment-216000593

> So, what about nicely requesting @pugjs to npm publish version 0.26.3, which is tagged on GitHub anyway, and pointing to this one on Mocha's package.json in a new patch release?
> It's not much of an improvement when you think of it, but compatibility is maintained and warning is removed...

Pretty please with sugar on top? :)

**Comment by jescalan** [comment]
At very least a stable release being published to `pug` would be fantastic. Right now jade produces a deprecation notice, but the current stable release of `pug` is an alpha 😕

**Comment by TimothyGu** [comment]
I'm not sure if there is a misunderstanding. We will NOT unpublish `jade` (the company has given us permission to continue to use that package name on npm, so long as new versions of `jade` are not being published), so you can in fact continue to use `jade` without any future repercussions.

Republishing old versions isn't trivial since the package name would have to be changed in `package.json`, and other unintended effects might result from the change.

Also, 0.26.3 is *kind of* ancient, like, it was published almost four years ago (!).

@jescalan, as I said in another issue, the notice is only a warning and doesn't really affect the user in any way. You can keep on using `jade`.

**Comment by jescalan** [comment]
Cool, thanks for the update @TimothyGu! Do you guys have any kind of extremely rough estimates as to when there will be a stable release of `pug`? I'm excited to upgrade!

**Comment by TimothyGu** [comment]
@jescalan, we do not. There are still a few things we want to get into Pug before a stable release, so no dates yet.

**Comment by SergioCrisostomo** [comment]
@TimothyGu (a bit offtopic here) but is there a _todo_ list so others can pick a line and help? Things like website, beta release or just renaming stuff/documentation?

**Comment by iknowcss** [comment]
@TimothyGu you're right, I did misunderstand. My reading of the legal letter you posted on another thread made me think that you had to unpublish everything with the jade moniker, including old versions. The warning is a minor annoyance but seeing lots of projects break abruptly would have been bad :)

Thanks!

**Comment by tomByrer** [comment]
IMHO the old Jade packages should stay inside their own ecosystem, but not breaking existing projects.  Is this the case now?

**Comment by iknowcss** [comment]
I agree. The old jade packages aren't going to disappear which was my original concern. The only issue which comes up now is the warning. I don't think publishing old jade versions under pug is the right fix for that anyway. 

Closing this issue since I don't see value in the change anymore. Thanks for the quick feedback!
